docker-compose.yaml

```diff
@@ -10,6 +10,7 @@ services:
     image: ankerctl/ankerctl:latest
     container_name: ankerctl
     restart: unless-stopped
+    pull_policy: build
```

Comment:
We talked about this in the last PR version, but can you elaborate again on what this does/fixes/improves?

1) Are there implications other than the (relatively niche) case of Portainer building from a git url?

2) What does this mean for people who try to build with `docker-compose` locally? Will it require internet access? Will it try to clone the repository again, if you do this on a fresh clone?

Reply:
To your second question: It will run the docker build command locally against the cloned source. This is probably ideal today since we're not publishing the container image to docker hub, but if we decide to do that later, then this will be a (mostly) redundant step.

I would prefer if we didn't set a pull policy (specifically if we decide to push this image to docker hub) as this is dependent on the user's setup.

Reply:
Does that mean, that if we push this to say, the github docker registry, and somebody uses this image in a `docker-compose.yml` file, the build process will try to make the image from git, instead of using the image we published?

Reply:
yes
https://docs.docker.com/compose/compose-file/05-services/#pull_policy

Reply:
`pull_policy: build` seems to be a somewhat recent addition to docker compose. 

Pros:
- Users running local development will always have the latest code baked into their docker containers

Cons:
- `docker compose up` will be slow because it will rebuild the container every time
- `ankerctl:latest` for me isn't necessarily `ankerctl:latest` for you (effectively bypassing the public image). This _shouldn't_ be a problem for most people, as long as that is understood when debugging/troubleshooting

It sort of comes down to the purpose of the docker compose. If it is to make it easy for end users to get ankerctl up and running, or if it is targeted twards developers who want to test inside a container

Alternatives:
1. Have 2 docker compose files, one for end users, and one for developers
- docker-compose.yaml
- docker-compose-dev.yaml

2. Pass in the build_arg as an optional argument. (Something like this):
```
docker compose up --build-arg PULL_POLICY=build
```
or
```
export COMPOSE_PULL_POLICY=build; docker compose up
```

Reply:
I can test to see if passing it as an optional `build_arg` works for me in Portainer, if not, then I can build images manually at CLI if I need to (if we see this as too much of a disruptive change). This also depends on whether or not there are ever going to be published container images and if so, how frequent these images will be updated.

Reply:
@billyjbryant Any news on this?

Since @Cisien got the github action PR merged, we are almost ready to publish images.

After listening to the inputs - and thank you all again for the detailed arguments - it's clear that this will not be merged as long as `pull_policy: build` is included.

We are trading a lot of potential issues and confusion, for a small convenience for some users. That's not worth it.

If there's some way to get convenience for everybody, I'm open to suggestions, but otherwise this needs a bit of rework.